Docs: Clean up Explore documentation

ifrost opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

This PR makes minor changes to the currently existing [Explore documentation](https://grafana.com/docs/grafana/latest/explore/). More information about changes is in the comments.

**Why do we need this feature?**

This is an existing and frequently used feature and should be well documented. This PR is just the first step to make the documentation more comprehensive. 

**Who is this feature for?**

Almost any Grafana user. Explore is a core feature, as important as Dashboards.

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes N/A

**Special notes for your reviewer**:

See more info in comments.

ifrost committed:
Remove the info about custom Prometheus Query Editor as it's not relevant since #60020

ifrost committed:
Update basic Explore documentation

ifrost committed:
Change graphs and tables to visualizations as Explore supports different types of visualization

ifrost committed:
Make clear "Starred" refers to a tab

ifrost committed:
Update how to select start/end date for query history

ifrost committed:
Remove Prometheus specific section as it's no longer relevant. Prometheus query editor is documented in https://grafana.com/docs/grafana/latest/datasources/prometheus/query-editor/#metrics-browser

ifrost committed:
Add informational log level (added in #30942)

ifrost committed:
Use more direct language

ifrost committed:
Fix a link

ifrost committed:
Make headers more consistent

ifrost committed:
Minor tweaks

ifrost commented on docs/sources/explore/_index.md:
```diff
@@ -44,12 +44,14 @@ To access Explore:
 
    {{< figure src="/static/img/docs/explore/panel_dropdown-7-4.png" class="docs-image--no-shadow" max-width= "650px" caption="Screenshot of the new Explore option in the panel menu" >}}
 
-1. Choose your data source from the dropdown in the top left. [Prometheus](https://grafana.com/oss/prometheus/) has a custom Explore implementation, the other data sources use their standard query editor.
```
Prometheus no longer has Explore-specific query editor (see #60020)

ifrost commented on docs/sources/explore/_index.md:
```diff
@@ -44,12 +44,14 @@ To access Explore:
    {{< figure src="/static/img/docs/explore/panel_dropdown-7-4.png" class="docs-image--no-shadow" max-width= "650px" caption="Screenshot of the new Explore option in the panel menu" >}}
 
-1. Choose your data source from the dropdown in the top left. [Prometheus](https://grafana.com/oss/prometheus/) has a custom Explore implementation, the other data sources use their standard query editor.
-1. In the query field, write your query to explore your data. There are three buttons beside the query field, a clear button (X), an add query button (+) and the remove query button (-). Just like the normal query editor, you can add and remove multiple queries.
```
The second bullet point is confusing, we don't even have these elements in the UI. The experience of writing a query in Explore is very much like in other places so I think we could simply cross-link with other docs.

Eve832 replied:
In the case of the steps, they should all be numbered `1.` our MD auto-enumerates. Also, I think starting with `2.` might confuse future contributors

ifrost replied:
Fixed in 26a300de5244a56147f23f3a551a5297dbc29d36

ifrost commented on docs/sources/explore/query-management.md:
```diff
@@ -73,29 +73,3 @@ You can customize the query history in the Settings tab. Options are described i
 
 > **Note:** Query history settings are global, and applied to both panels in split mode.
-
-## Prometheus-specific Features
```
We don't have Prometheus-specific features anymore. These features are now available also in Dashboards and documented in Prometheus docs: https://grafana.com/docs/grafana/latest/datasources/prometheus/query-editor/#metrics-browser

ifrost commented on docs/sources/explore/trace-integration.md:
```diff
@@ -20,9 +20,9 @@ Supported data sources are:
 For information on how to configure queries for the data sources listed above, refer to the documentation for specific data source.
 
-## Trace View explanation
+## Trace View
```
Just to make headers in this page more consistent

ifrost commented on docs/sources/explore/trace-integration.md:
```diff
@@ -20,9 +20,9 @@ Supported data sources are:
-## Trace View explanation
+## Trace View
 
-This section explains the elements of the Trace View dashboard.
```
Trace View is a quite complex visualization, but calling it a dashboard sounds confusing.

Eve832 commented on docs/sources/explore/explore-inspector.md:
```diff
@@ -43,7 +43,7 @@ For multiple queries or for queries multiple nodes, there are additional options
 Grafana generates a CSV file in your default browser download location. You can open it in the viewer of your choice.
 
-1. In the **Inspector** tab, raw query results following instructions in (### Inspect raw query results).
+1. In the **Inspector** tab, raw query results following instructions [described above](#inspect-raw-query-results).
```
```suggestion
1. In the **Inspector** tab, get raw query results by following the instructions [described above](#inspect-raw-query-results).
```

Eve832 commented on docs/sources/explore/trace-integration.md:
```diff
@@ -79,7 +79,7 @@ You can navigate from a span in a trace view directly to logs relevant for that
 Click the document icon to open a split view in Explore with the configured data source and query relevant logs for the span.
 
-## Service Graph view
+## Service Graph View
```
If it's not capitalized in the text, it shouldn't be capitalized in the heading. Instinct says we shouldn't refer to Service Graph View, but rather Service graph view and Node graph view, though I'm guessing they're capitalized in the UI and that's why you're making these edits..

ifrost replied:
That's a good point. We refer to "Trace View" as the full name of the component displaying trace details. In case of "Node Graph" and "Service Graph" we don't really use "view" as the word "graph" indicates type of visualization. 

Maybe it would make more sense to use: Trace View, Service Graph and Node Graph? Also use capitalized "view" only in "Trace View"?

ifrost replied:
Updated in 0d5dedc7d1e1ec378e256e2cbb6e4306379acc28

ifrost committed:
Update docs/sources/explore/explore-inspector.md

Co-authored-by: Eve Meelan <81647476+Eve832@users.noreply.github.com>

ifrost committed:
Merge branch 'main' into ifrost/explore-docs-clean-up

ifrost committed:
Do not enumerate bullet points

ifrost committed:
Fix incorrect header type

ifrost committed:
Update tracing headers

ifrost committed:
Link to more info about Mixed data source

ifrost committed:
Use "data source" instead of "datasource"

ifrost committed:
Fix tracing headers

ifrost committed:
Fix more log headers

ifrost commented:
A few more changes added to the PR:

* Fixes to incorrect headers on Logs page: 60974f4dc3a3aab2da3d5ad2e5108f165bfa0e03 and 35094e0de3c6b4ada8e85cb7bc9c00b7ed831204
* Fixed to incorrect headers on Tracing page: db2d98c396d6769d7a327b69754588ec4f3e8a51
* Cross link to more info about Mixed data source: 438ae9fc70bbc6fe2c6590626626f35e933e5245
* Changed "datasource" to "data source":  c5ae7d5bb0bb3486973c9789f57c1c0913c77bee

ifrost commented:
@grafana/docs-squad, I have updated the PR with suggested changes. Could you please have another look at it? 🙏 Thanks!

JStickler commented on docs/sources/explore/explore-inspector.md:
```diff
@@ -43,7 +43,7 @@ For multiple queries or for queries multiple nodes, there are additional options
 Grafana generates a CSV file in your default browser download location. You can open it in the viewer of your choice.
 
-1. In the **Inspector** tab, raw query results following instructions in (### Inspect raw query results).
+1. In the **Inspector** tab, get raw query results by following the instructions [described above](#inspect-raw-query-results).
```
```suggestion
1. In the **Inspector** tab, get raw query results by following the instructions [described in Inspect raw query results](#inspect-raw-query-results).
```

JStickler reviewed (approved):
[docs team] LGTM

ifrost committed:
Update docs/sources/explore/explore-inspector.md

Co-authored-by: J Stickler <julie.stickler@grafana.com>
